### DISCUSSION TITLE
How to disable the enlarge when I make a magnified gesture on the touchpad

### DISCUSSION THREAD

**Discussion by linesoft2** [tauri-apps/tauri#3843] [Q&A] [answered]
On Windows

**Comment by FabianLars** [comment, marked as answer]
For reference, this is the function to disable it: https://docs.rs/webview2-com/latest/webview2_com/Microsoft/Web/WebView2/Win32/struct.ICoreWebView2Settings5.html#method.SetIsPinchZoomEnabled

Users should be able to use that themselves once/if we expose https://docs.rs/wry/latest/x86_64-pc-windows-msvc/wry/webview/struct.WebView.html#method.controller in tauri.

But i feel like we should match this setting with the default zoom functionality (which is currently always disabled.).

**Comment by Jethril** [comment]
There is an environment variable that can be used: `WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS` with arguments `--disable-pinch`. You may also be interested in `--disable-features=OverscrollHistoryNavigation,msExperimentalScrolling`, which disables the "previous" and "next" gestures.

There is the same problem with WebkitGtk on Linux, and the only way I found to disable it is highly unsafe:
```rs
/// A Tauri plugin that customizes some WebKitGTK behavior.
pub struct WindowCustomizerPlugin;

impl Default for WindowCustomizerPlugin {
    fn default() -> Self {
        Self
    }
}

impl<R: Runtime> Plugin<R> for WindowCustomizerPlugin {
    fn name(&self) -> &'static str {
        "Zoom disabler plugin"
    }

    fn created(&mut self, window: Window<R>) {
        let gtk_window = window.gtk_window().expect("GTK window should be accessible");

        // Retrieve the web view.
        let web_view = gtk_window
            .children()
            .into_iter()
            .next()
            .expect("GTK window should have a child")
            .downcast::<gtk::Box>()
            .expect("GTK window child should be a box")
            .children()
            .into_iter()
            .filter_map(|w| w.downcast::<WebView>().ok())
            .next()
            .expect("box should have a webview child");

        // This piece of code is responsible for preventing the context-menu (right-click menu) from being shown.
        web_view.connect_context_menu(|_, _, _, _| true);

        unsafe {
            // Disable the gesture zoom by removing its handlers.
            if let Some(data) = web_view.data::<GestureZoom>("wk-view-zoom-gesture") {
                gobject_ffi::g_signal_handlers_destroy(data.as_ptr().cast());
            }

            // Disable some gestures by freeing their controllers.
            // This won't work with GTK 4, we would use the controller API for this.
            web_view.set_data("wk-view-long-press-gesture", ());
            web_view.set_data("wk-view-swipe-gesture", ());
        }
    }
}
```

Shouldn't we open an issue for this @FabianLars ?

**Reply by linesoft2** [reply]
Does this PR solve this problem?
https://github.com/tauri-apps/wry/pull/896/commits

**Reply by FabianLars** [reply]
One thing to simplify your code is to use `window.with_webview(|webview| {})` It will give you direct access to the webview.

> Shouldn't we open an issue for this FabianLars ?

Doesn't this one count? https://github.com/tauri-apps/wry/issues/544

Also one thing i saw on the internet for wkwebview (the webkitgtk equivalent on macos) is that they specified a `maximum-scale` on the meta html element. Could someone try that out? example: https://stackoverflow.com/questions/40452034/disable-zoom-in-wkwebview/41741125#41741125 (you can do it directly in your html file)

**Reply by Jethril** [reply]
> One thing to simplify your code is to use `window.with_webview(|webview| {})` It will give you direct access to the webview.

Oh, I'll try that, thanks.

> Doesn't this one count? [tauri-apps/wry#544](https://github.com/tauri-apps/wry/issues/544)

Sure, but it is Linux-only. I could also edit my issue...

> Also one thing i saw on the internet for wkwebview (the webkitgtk equivalent on macos) is that they specified a `maximum-scale` on the meta html element. Could someone try that out? example: https://stackoverflow.com/questions/40452034/disable-zoom-in-wkwebview/41741125#41741125 (you can do it directly in your html file)

I tried, yes. The zoom is directly handled by GTK and completely ignores any in-browser feature.

**Comment by IlimaVIP** [comment]
Any updates on this issue @Jethril @FabianLars

**Reply by Jethril** [reply]
I did not try to reproduce the issues since I'm no longer working on the subject in my company but I could do that on my spare time if this is needed.

**Comment by niraj-khatiwada** [comment]
How can we disable this on Linux? The JavaScript side cannot do this, so there must be a way in the webview (maybe webview args) to do this. The two fingure zoom always enlarges the view no matter what:

https://github.com/tauri-apps/tauri/assets/51436631/13ac81d4-feae-4a81-b9ea-62a248ee9278
